### DISCUSSION TITLE
[How to] Use Client Events in Livewire

### DISCUSSION THREAD

**Discussion by cjmellor** [livewire/livewire#5476] [2. Show and tell]
Hi,

Recently, I have been messing around, learning how to use Broadcasting in Laravel, and I ended up building a chat room. I originally did this using Javascript (AlpineJS) but I wanted to see how well it would work using Livewire, as it has built-in tools to use Echo.

One thing I couldn't figure out, and couldn't even find a solution online (not even in the Docs), was how to use Client Events, say if you wanted a "User is typing" indicator.

Turns out it is pretty darn easy 😅 you just prefix the event with `.client-`

**Code example:**

```php
"echo-presence:chat-room,.client-typing" => "typing",
"echo-presence:chat-room,.client-stopped-typing" => "stoppedTyping",
```

Then create the `typing` and `stoppedTyping` methods and perform your logic

```php
public function typing($event)
{
    //
}

public function stoppedTyping($event)
{
    //
}
```

One thing to note (and unless I am mistaken) you still have to listen for the events using Javascript, for the broadcast provider to get the data. What data you send is up to you -- the below example sends the ID of the user who is typing.

```js
const whisperEventName = value === "" ? "stopped-typing" : "typing";

Echo.join("chat-room").whisper(whisperEventName, {
    id: {{ auth()->id() }}
});
```

_Event and channel names will vary to your own preferences._

**This is how I got it to work... there may be different ways to do it.**

**Comment by hasan-aftab** [comment]
It worked for me Thanks brother @cjmellor

**Comment by Ghazanfar-Sayed** [comment]
in case of listenForWhisper to a private channel, you should instead of the event name put the whisper event with prefix of .client-
something like this:
"echo-private:chat-room,.client-typing" => "typing",
"echo-private:chat-room,.client-stopped-typing" => "stoppedTyping",

thanks @cjmellor
